iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessor.java

```diff
@@ -1323,11 +1348,11 @@ private void switchWaitingToWorking()
     LOGGER.info("The status of filenode processor {} switches from {} to {}.", getProcessorName(),
         FileNodeProcessorStatus.WAITING, FileNodeProcessorStatus.NONE);
 
-    if (oldMultiPassLock != null) {
-      LOGGER.info("The old Multiple Pass Token set is {}, the old Multiple Pass Lock is {}",
+    if (oldMultiPassCount != null) {
+      LOGGER.info("The old Multiple Pass Token set is {}, the old Multiple Pass Count is {}",
           oldMultiPassTokenSet,
-          oldMultiPassLock);
-      oldMultiPassLock.writeLock().lock();
+          oldMultiPassCount);
+      oldMultiPassLock.lock();
```

Comment:
original code `oldMultiPassLock.writeLock().lock();` is used to check that all old queries are over.
If all queries are over, we can delete invalid tsfile and overflowfile.
New your changes can't make sure that corner case.

Reply:
Does `oldMultiPassLock.lock();` not wait for getting the lock?

Reply:
Well, unfortunately, it's a bug. I change to use CountDownLatch instead of AtmicLong for oldMultiPassCount because oldMultiPassCount will never increase query token num and it just countdown and use CountDownLatch.await() to wait until all old queries are over